### DISCUSSION TITLE
Operator showing OOM warnings/errors

### DISCUSSION THREAD

**Discussion by saloni-2404** [strimzi/strimzi-kafka-operator#4108] [General]
How can we better configure the operator to avoid the following issue?

2020-12-03T16:47:27.089+0000: 0.104: [GC (Allocation Failure) 2020-12-03T16:47:27.089+0000: 0.104: [DefNew: 2176K->255K(2432K), 0.0015106 secs] 2176K->448K(7936K), 0.0015588 secs] [Times: user=0.00 sys=0.00, real=0.00 secs]  and occasionally it dies via OOM

Strimzi operator runs out of memory with the default config,  looks like  ~100 restarts per day

NAME                                                           READY STATUS                       RESTARTS   AGE
----------------------------------------------------------------------------------------------------- 
strimzi-operator-7fc59d96f-nssp5   1/1     Running                        1110             10d
strimzi-operator-7fc59d96f-nssp5   0/1     OOMKilled                   1110             10d
strimzi-operator-7fc59d96f-nssp5   0/1     CrashLoopBackOff   1110             10d
strimzi-operator-7fc59d96f-nssp5   0/1     Running                        1111            10d

**Comment by scholzj** [comment]
How many resources do you have and what operator version are you using?

In general, you can try to give it more resources by editing the cluster-operator deployment. You can also use the environment variable  `JAVA_OPTS` to pass it some specific Java options such as `-Xmx` etc.

**Comment by saloni-2404** [comment]
Operator version is: 2.2.20
Resources- it was having this problem on a small cluster : 3 brokers 3 zk 1 MM 1 MM2

**Reply by scholzj** [reply]
There is no 2.2.20 ... id you mean 0.20.0 I guess? That is weird. But as I suggested above, you can try to tune the resources.

**Comment by saloni-2404** [comment]
Sorry, I meant 0.17.0 version

**Reply by scholzj** [reply]
Ok. We for sure did some improvements to the memory handling in 0.20.0. So it might be actually better in the latest version.

**Comment by saloni-2404** [comment]
There are two use cases- 
a) smaller cluster 4 brokers 3 zk, monitoring and not much else. 
b) large cluster 12 brokers , 3zk 1x mm 3x mm2 and monitoring
How much resources should we be ideally giving in deployment-operator.yaml?
Current one is:
resources:
            limits:
              cpu: 1000m
              memory: 256Mi
            requests:
              cpu: 200m
              memory: 256Mi

**Reply by scholzj** [reply]
I think we configure the operator now with 384Mi or 512Mi

**Reply by saloni-2404** [reply]
ok thanks Jakub
